pkg/apis/common/resource.go

```diff
@@ -0,0 +1,30 @@
+package common
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+// PartialResource represents a linked resource in GS
+type PartialResource struct {
+	Identifier string `json:"identifier,omitempty"`
+	Name       string `json:"name,omitempty"`
+}
+
+// MarshalJSON unfolds the resource to its identifier
+func (pr PartialResource) MarshalJSON() ([]byte, error) {
+	return json.Marshal(pr.Identifier)
+}
+
+// PartialResourceList represents a linked resource list in GS
+type PartialResourceList []PartialResource
```

Comment:
this thing is needed for resource lists represented in the API as a single string, maybe rename the type to `PartialResourceListString` or something?

IIRC there are other cases where the API works with a `[]PartialResource` as expected and I want to prevent people being confused they should always use this type - hence the slightly more specific name

Reply:
Not sure if `PartialResourceListString` is a better name for this type as it only encodes to (not decodes from) string. Maybe we should move this type to the `gs` sub-pkg?

Reply:
I think it's *technically* not related to GS, so moving it in the `gs` package feels wrong to me

Any other ideas how to call it, given my reason to rename it above?

Reply:
AFAIK encoding to comma separated identifiers is unique to GS. The pyengine also has similar logic in a file called `generic_service_resource.py`. Can you point me to an engine endpoint, which also uses this?

I don't think we should make a (more) general purpose type out of this as it has a specialized Marshaler, and therefore still think it'l be best placed within the gs package.